src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -89,9 +126,27 @@ public SocksBeginExBuilder typeId(
         }
 
         public SocksBeginExBuilder address(
-            String address)
+            String address) throws UnknownHostException
         {
-            this.beginExRW.address(address);
+            if (IPV4_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final Matcher ipv4Matcher = IPV4_ADDRESS_MATCHER.get();
+                final byte[] ipv4AddressBytes = IPV4_ADDRESS_BYTES.get();
+                for (int i = 0; i < ipv4AddressBytes.length; i++)
+                {
+                    ipv4AddressBytes[i] = (byte) Integer.parseInt(ipv4Matcher.group(i + 1));
+                }
+                beginExRW.address(b -> b.ipv4Address(s -> s.set(ipv4AddressBytes)));
+            }
+            else if (IPV6_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final byte[] addressBytes = InetAddress.getByName(address).getAddress();
```

Comment:
Next let's eliminate the garbage from this code for `IPv6` case.
Please propose a solution similar to (but not exactly the same as) how we solved for `IPv4` case.